File: DynamicProgramming/338_CountingBits.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        if num < 0:
            return []
        ret = [0] * (num + 1)
        if num >= 1:
            ret[1] = 1
            for x in range(2, num+1):
                d = x/2
                r = x%2
                ret[x] = ret[d] + ret[r]
        return ret

File: DynamicProgramming/338_CountingBits.java
public class Solution {
    public int[] countBits(int num) {
        if (num < 0) {
            return new int[0];
        }

        int[] ret = new int[num + 1];
        ret[0] = 0;
        if (num > 0) {
            ret[1] = 1;
            for (int i=2; i<num+1; i++) {
                int d = i/2;
                int r = i%2;
                ret[i] = ret[d] + ret[r];
            }
        }
        return ret;
    }
}